Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM ubuntu:xenial
+
+# Default values for the build
+ARG git_branch
+ARG git_slug
+ARG c_compiler
+ARG cxx_compiler
+ARG impl
+
+RUN apt-get -yq update
+
+# Utilities
+RUN apt-get install -yq --allow-downgrades --allow-remove-essential            \
+    --allow-change-held-packages git wget apt-utils cmake unzip                \
+    libboost-all-dev software-properties-common python-software-properties libcompute-dev
+
+RUN add-apt-repository -y ppa:ubuntu-toolchain-r/test
+
+RUN apt-get -yq update
+
+# Clang 4.0
+RUN if [ "${c_compiler}" = 'clang-4.0' ]; then apt-get install -yq             \
+    --allow-downgrades --allow-remove-essential --allow-change-held-packages   \
+     clang-4.0 libomp-dev; fi
+
+# GCC 6
+RUN if [ "${c_compiler}" = 'gcc-6' ]; then apt-get install -yq                 \
+    --allow-downgrades --allow-remove-essential --allow-change-held-packages   \
+    g++-6 gcc-6; fi
+
+# OpenCL
```

Comment:
note this pulls only the ICD loader, not an OpenCL implementation

Reply:
Boost.compute (libcompute-dev) also actually pulls them in as it needs the headers.